### PR TITLE
docs: initial `resource` docs

### PR DESCRIPTION
**Pull request by alxhub**
Adding an initial pass of docs for `resource` under the Signals heading.

### CODE REVIEW

File: adev/src/content/guide/signals/resource.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+const userResource = resource({
+  request: () => ({id: userId()}),
+  loader: ({request}) => fetchUser(request),
+});
+
+// ...
+
+userResource.reload();
+```
```
Comment (jelbourn) [inline adev/src/content/guide/signals/resource.md:86]: @alxhub one thing I wasn't sure about— do we want to give any advice here on circumstances under which it makes sense to manually reload the resource?

File: adev/src/content/guide/signals/resource.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+### `undefined` requests
+
+A request value of `undefined` prevents the resource from running its loader, and will put the resource into an `Idle` state.
```
Comment (jelbourn) [inline adev/src/content/guide/signals/resource.md:90]:
```suggestion
A request value of `undefined` prevents the resource from running its loader and puts the resource into an `Idle` state.
```

**Review by jelbourn — APPROVED** [review]
LGTM

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit cd4907d93c5cb043cee3f3b9de3f64e811ec68dd.

The changes were merged into the following branches: main, 19.0.x

**Comment by AndrewKushnir** [comment]
@alxhub FYI, I've merged this PR as is for now, please create a followup PR if needed, based on Jeremy's feedback above.

**Pull request closed without merging**